aws/resource_aws_appsync_graphql_api_test.go

```diff
@@ -0,0 +1,112 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appsync"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsAppsyncGraphqlApi_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsAppsyncGraphqlApiDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAppsyncGraphqlApiConfig_apikey(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncGraphqlApiExists("aws_appsync_graphql_api.test_apikey"),
+					resource.TestCheckResourceAttrSet("aws_appsync_graphql_api.test_apikey", "arn"),
+				),
+			},
+			{
+				Config: testAccAppsyncGraphqlApiConfig_iam(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncGraphqlApiExists("aws_appsync_graphql_api.test_iam"),
+					resource.TestCheckResourceAttrSet("aws_appsync_graphql_api.test_iam", "arn"),
+				),
+			},
+			{
+				Config: testAccAppsyncGraphqlApiConfig_cognito(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncGraphqlApiExists("aws_appsync_graphql_api.test_cognito"),
+					resource.TestCheckResourceAttrSet("aws_appsync_graphql_api.test_cognito", "arn"),
+				),
+			},
+		},
+	})
+}
```

Comment:
Do you mind separating out the 3 steps into 3 tests? This will allow us to run them in parallel more easily. You can keep `testAccAppsyncGraphqlApiConfig_apikey` in `_basic` here, just to stick to the convention.